File: spec/factories/enabled_dynamic_fields.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :enabled_dynamic_field do
    association :project
    association :dynamic_field

    field_sets { [] }
    digital_object_type { 'item' }
    required            { true }
    locked              { false }
    hidden              { false }
    owner_only          { false }
    shareable           { false }
    default_value       {}
  end
end

File: app/javascript/src/hyacinth_v1/graphql/projects/enabledDynamicFields.js
import gql from 'graphql-tag';

const getEnabledDynamicFieldsFields = `
project {
  stringKey
}
digitalObjectType
`;

export const getEnabledDynamicFieldsQuery = gql`
  query($project: StringKey!, $digitalObjectType: DigitalObjectTypeEnum!) {
    enabledDynamicFields(project: $project, digitalObjectType: $digitalObjectType) {
      type: __typename
      project {
        stringKey
      }
      digitalObjectType
      dynamicField {
        id
      }
      fieldSets {
        id
      }
      enabled
      required
      locked
      hidden
      ownerOnly
      defaultValue
      shareable
    }
  }
`;

export const updateEnabledDynamicFieldsMutation = gql`
  mutation ($input: UpdateProjectEnabledFieldsInput!) {
    updateProjectEnabledFields(input: $input) {
      projectEnabledFields {
        ${getEnabledDynamicFieldsFields}
        dynamicField {
          id
        }
        fieldSets {
          id
        }
        required
        locked
        hidden
        ownerOnly
        defaultValue
        shareable
      }
      userErrors {
        message
        path
      }
    }
  }
`;
